tsdb/config.go

```diff
@@ -77,8 +81,19 @@ const (
 	// partition snapshot compactions that can run at one time.
 	// A value of 0 results in runtime.GOMAXPROCS(0).
 	DefaultSeriesFileMaxConcurrentSnapshotCompactions = 0
+
+	// MaxTSMFileSize is the maximum size of TSM files.
+	MaxTSMFileSize = uint32(2048 * 1024 * 1024) // 2GB
+
 )
 
+// SingleGenerationReason outputs a log message for our single generation compaction
+// when checked for full compaction.
+// 1048576000 is a magic number for bytes per gigabyte.
+func SingleGenerationReason() string {
+	return fmt.Sprintf("not fully compacted and not idle because single generation with many files under %d GB and many files under aggressive compaction points per block count (%d points)", int(MaxTSMFileSize/1048576000), AggressiveMaxPointsPerBlock)
+}
+
```

Comment:
If this is only used once, it probably doesn't make sense to break it into a function. 

Also, "many" doesn't help me understand why the compaction isn't scheduled. Is there a more concrete way to explain the scheduling decision to the user?

Reply:
This is essentially a constant, so you could make a string var and get the same effect. When I see the function call, I thought maybe there was a side-effect to it.

Reply:
Can I use SprintF in a const variable though?

Reply:
This was my request, to ensure the message captures any changes to the numeric constants, which seem possible, given this PR.

Sprintf cannot be used to set constants.  The value (which is reused in many tests) needs to be a variable.